src/pages/RoomInvitePage.tsx

```diff
@@ -184,6 +184,14 @@ function RoomInvitePage({
         Navigation.navigate(backRoute);
     }, [selectedOptions, backRoute, reportID, validate]);
 
+    const goBack = useCallback(() => {
+        if (role === CONST.IOU.SHARE.ROLE.ACCOUNTANT) {
+            Navigation.dismissModal(reportID);
+            return;
+        }
+        Navigation.goBack(backRoute);
+    }, [role, reportID, backRoute]);
+
```

Comment:
Added this fix, otherwise we will have infinite back loop as shown below

https://github.com/Expensify/App/assets/50919443/bcc064ab-0cdb-45a8-9a54-842bf11e0322